src/pages/Editor.vue

```diff
@@ -602,35 +666,169 @@ export default {
       this.showDialogBox = true;
     },
     dialogConfirmed() {
       // invoked when the confirm button of the dialog box is clicked
       // update the dialog properties
       this.dialogConfirmButtonConfig = {
+        enabled: true,
         class: "hidden",
       };
       this.dialogCancelButtonConfig = {
+        enabled: true,
         class: "hidden",
       };
       this.dialogDescription = "";
-      this.dialogTitle = this.publishInProgressDialogTitle;
-      // publish the plio or its changes
-      this.publishPlio();
+
+      // call separate methods depening on the dialog action that
+      // was set
+      if (this.dialogAction == "publish")
+        this.dialogPublish()
+      else if (this.dialogAction == "deleteItem")
+        this.dialogDeleteItem()
+      else if (this.dialogAction == "closeDialog")
+        this.showDialogBox = false
+
+      // reset the dialog action value
+      this.dialogAction = ""
     },
     dialogCancelled() {
       // invoked when the cancel button of the dialog box is clicked
       this.showDialogBox = false;
     },
+    showCannotAddItemDialog() {
+      // set up the dialog properties when user tries to add an item
+      // at an invalid time
+      this.dialogTitle = "Cannot add a new question here";
+      this.dialogDescription = "Questions should be at least 2 seconds apart. Please choose a different time for the question";
+      this.dialogConfirmButtonConfig = {
+        enabled: true,
+        text: "Got it!",
+        class: "bg-primary-button hover:bg-primary-button-hover focus:outline-none focus:ring-0",
+      };
+      this.dialogCancelButtonConfig = {
+        enabled: false,
+        text: "",
+        class: ""
+      };
+
+      // carry out the closeDialog action when dialog is closed
+      this.dialogAction = "closeDialog"
+      // show the dialogue
+      this.showDialogBox = true;
+    },
+    dialogPublish() {
+      this.dialogTitle = this.publishInProgressDialogTitle;
+      // publish the plio or its changes
+      this.publishPlio();
+    },
+    dialogDeleteItem() {
+      this.dialogTitle = this.deleteItemInProgressDialogTitle;
+      // delete the selected item after user confirms
+      this.deleteSelectedItem();
+    },
+    getTimestampForNewItem() {
+      // loop through itemTimestamps to check if the time where the user
+      // is trying to add an item is valid or not
+      // using for loop instead of forEach as forEach was running async
+
+      // returning -1 to signify that the time chosen by the user is not valid
+      for (let index = 0; index < this.itemTimestamps.length; index++) {
+        var val = this.itemTimestamps[index]
+        if (val == this.currentTimestamp) return -1
+        if ( this.currentTimestamp <= (val + 2) && this.currentTimestamp >= (val - 2) )
+          return -1
+      }
+      return this.currentTimestamp
+    },
+    getItemTypeForNewItem() {
+      // hardcoded for now
+      return "question"
+    },
```

Comment:
we should not do a generic item for the Add Question button - for other Items there will be separate buttons - so, won't have to mention hardcoded then

also, 1-liner.